### PR TITLE
SSS: make dmfp inputs scalars

### PR DESCRIPTION
**Pull request by est77**

### CODE REVIEW

File: src/appleseed/renderer/modeling/bssrdf/normalizeddiffusionbssrdf.cpp [hunk 1]
```diff
@@ -114,13 +114,8 @@ namespace
...
 
-            if (values->m_dmfp.size() != values->m_reflectance.size())
-            {
-                if (values->m_dmfp.is_spectral())
-                    Spectrum::upgrade(values->m_reflectance, values->m_reflectance);
-                else
-                    values->m_reflectance = values->m_reflectance.convert_to_rgb(*m_lighting_conditions);
-            }
+            // Clamp reflectance.
+            values->m_reflectance = clamp(values->m_reflectance, 0.0f, 1.0f);
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bssrdf/normalizeddiffusionbssrdf.cpp]: `saturate()`

File: src/appleseed/renderer/modeling/bssrdf/directionaldipolebssrdf.cpp [hunk 2]
```diff
@@ -122,15 +122,10 @@ namespace
...
 
-            if (values->m_dmfp.size() != values->m_reflectance.size())
-            {
-                if (values->m_dmfp.is_spectral())
-                    Spectrum::upgrade(values->m_reflectance, values->m_reflectance);
-                else
-                    values->m_reflectance = values->m_reflectance.convert_to_rgb(*m_lighting_conditions);
-            }
+            // Clamp reflectance.
+            values->m_reflectance = clamp(values->m_reflectance, 0.0f, 1.0f);
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bssrdf/directionaldipolebssrdf.cpp]: `saturate()`

File: src/appleseed/renderer/modeling/bssrdf/sss.h [hunk 3]
```diff
@@ -95,7 +95,7 @@ double reduced_extinction_coefficient(
 
 void compute_absorption_and_scattering(
     const Spectrum& rd,                     // surface albedo
-    const Spectrum& dmfp,                   // diffuse mean free path
+    const double& dmfp,                     // diffuse mean free path
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bssrdf/sss.h]: Pass by value. Fix indentation.

File: src/appleseed/renderer/modeling/bssrdf/sss.cpp [hunk 4]
```diff
@@ -117,14 +117,12 @@ double reduced_extinction_coefficient(
 
 void compute_absorption_and_scattering(
     const Spectrum& rd,
-    const Spectrum& dmfp,
+    const double&   dmfp,
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bssrdf/sss.cpp]: Pass by value.

Comment (est77) [inline src/appleseed/renderer/modeling/bssrdf/sss.cpp]: Fixed and rebased.

File: src/appleseed/renderer/modeling/bssrdf/bssrdf.cpp [hunk 5]
```diff
@@ -31,7 +31,6 @@
 
 // appleseed.renderer headers.
 #include "renderer/kernel/shading/shadingpoint.h"
-#include "renderer/modeling/frame/frame.h"
 #include "renderer/modeling/input/inputevaluator.h"
 #include "renderer/modeling/project/project.h"
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bssrdf/bssrdf.cpp]: You can also remove `project.h`.

**Pull request merged**